gpkit/varkey.py

```diff
@@ -70,55 +70,39 @@ def __setstate__(self, state):
         "Restores varkey from its metadata dictionary"
         self.__init__(**state)
 
-    def str_without(self, excluded=None):
-        "Returns string without certain fields (such as 'models')."
-        if excluded is None:
-            excluded = []
+    def str_without(self, excluded=()):
+        "Returns string without certain fields (such as 'lineage')."
         string = self.name
         for subscript in self.subscripts:
             if self.descr.get(subscript) and subscript not in excluded:
                 substring = self.descr[subscript]
-                if subscript == "models":
-                    if self.modelnums and "modelnums" not in excluded:
-                        substring = ["%s.%s" % (ss, mn) if mn > 0 else ss
-                                     for ss, mn
-                                     in zip(substring, self.modelnums)]
-                    substring = "/".join(substring)
+                if subscript == "lineage":
+                    substring = self.lineagestr("modelnums" not in excluded)
                 string += "_%s" % (substring,)
         return string
 
     def __getattr__(self, attr):
         return self.descr.get(attr, None)
 
-    unitstr = unitstr
+    @property
+    def models(self):
```

Comment:
call it `modelnames`, to make it clear that it's not returning `Model` objects? I don't feel strongly about this one though.